pkg/addons/ensure.go

```diff
@@ -106,20 +115,41 @@ func collectAddons(s *state.State) (addonsToDeploy []addonAction) {
 		})
 	}
 
+	// Special case for AWS
+	if s.Cluster.CloudProvider.AWS != nil {
+		if s.Cluster.CloudProvider.External {
+			addonsToDeploy = append(addonsToDeploy,
+				addonAction{
+					name: resources.AddonCCMAws,
+				},
+				addonAction{
+					name: resources.AddonCSIAwsEBS,
+				},
+			)
+		} else {
+			// since k8s 1.23 CSIMigrationAWS is turn by default and require installation of AWS EBS CSI driver even if
+			// we don't use external ccm (https://github.com/kubernetes/kubernetes/pull/106098)
+			k8sVersion, err := semver.NewVersion(s.Cluster.Versions.Kubernetes)
```

Comment:
IMO, we can use `MustParse` instead because validation ensures that it's a valid semver, so this can't fail.